packages/orbit-components/src/Timeline/TimelineContext.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import * as React from "react";
+
+import type { StepContext, Context } from "./TimelineContext";
+
+const initialState = {
+  statuses: [],
+};
+
+export const TimelineStatusContext: React.Context<Context> = React.createContext({
+  state: initialState,
+  dispatch: () => null,
+});
+
+export const TimelineStepContext: React.Context<StepContext> = React.createContext({
+  index: 0,
+  last: false,
+});
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_STATUS":
+      return {
+        statuses: [...state.statuses, action.payload],
+      };
+
+    default:
+      return state;
+  }
+};
+
+export const TimelineStatusProvider = ({ children }: { children: React.Node }) => {
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+
+  return (
+    <TimelineStatusContext.Provider value={{ state, dispatch }}>
+      {children}
+    </TimelineStatusContext.Provider>
+  );
+};
```

Comment:
`useState` seems to be enough here, this way you can offload most of the logic to the `TimelineStep` component so it's less split:

```suggestion
export const TimelineStatusProvider = ({ children }: { children: React.Node }) => {
  const [statuses, setStatuses] = React.useState([]);

  return (
    <TimelineStatusContext.Provider value={{ statuses, setStatuses }}>
      {children}
    </TimelineStatusContext.Provider>
  );
};
```

Reply:
good point, thx, my approach was a bit over-engineered :D